lib/pages/timetable/view/events/add_event_view.dart

```diff
@@ -1,765 +1,792 @@
-//
-// extension LocalDateComparisons on LocalDate {
-//   bool isDuring(AllDayUniEvent semester) {
-//     return DateInterval(semester.startDate, semester.endDate).contains(this);
-//   }
-//
-//   bool isBeforeOrDuring(AllDayUniEvent semester) {
-//     if (compareTo(semester.startDate) < 0) return true;
-//     return isDuring(semester);
-//   }
-// }
+import 'package:dart_date/dart_date.dart' show Interval;
+import 'package:dotted_line/dotted_line.dart';
+import 'package:flutter/material.dart' hide Interval;
+import 'package:flutter/rendering.dart';
+import 'package:flutter_feather_icons/flutter_feather_icons.dart';
+import 'package:provider/provider.dart';
+import 'package:rrule/rrule.dart';
+import 'package:time_machine/time_machine.dart' as time_machine show DayOfWeek;
+import 'package:time_machine/time_machine.dart' hide Interval;
+import 'package:time_machine/time_machine_text_patterns.dart';
+
+import '../../../../authentication/model/user.dart';
+import '../../../../authentication/service/auth_provider.dart';
+import '../../../../generated/l10n.dart';
+import '../../../../navigation/routes.dart';
+import '../../../../resources/custom_icons.dart';
+import '../../../../resources/locale_provider.dart';
+import '../../../../widgets/button.dart';
+import '../../../../widgets/dialog.dart';
+import '../../../../widgets/scaffold.dart';
+import '../../../../widgets/selectable.dart';
+import '../../../../widgets/toast.dart';
+import '../../../classes/model/class.dart';
+import '../../../classes/service/class_provider.dart';
+import '../../../filter/service/filter_provider.dart';
+import '../../../filter/view/relevance_picker.dart';
+import '../../../people/model/person.dart';
+import '../../../people/service/person_provider.dart';
+import '../../../people/view/people_page.dart';
+import '../../model/academic_calendar.dart';
+import '../../model/events/all_day_event.dart';
+import '../../model/events/class_event.dart';
+import '../../model/events/recurring_event.dart';
+import '../../model/events/uni_event.dart';
+import '../../service/uni_event_provider.dart';
+import '../../timetable_utils.dart';
+
+class AddEventView extends StatefulWidget {
+  /// If the `id` of [initialEvent] is not null, this acts like an "Edit event"
+  /// page starting from the info in [initialEvent]. Otherwise, it acts like an
+  /// "Add event" page with optional default values based on [initialEvent].
+  const AddEventView({Key key, this.initialEvent}) : super(key: key);
+
+  final UniEvent initialEvent;
+
+  @override
+  _AddEventViewState createState() => _AddEventViewState();
+}
+
+class _AddEventViewState extends State<AddEventView> {
+  final formKey = GlobalKey<FormState>();
+
+  TextEditingController locationController;
+  RelevanceController relevanceController = RelevanceController();
+
+  UniEventType selectedEventType;
+  ClassHeader selectedClass;
+  Person selectedTeacher;
+  String selectedCalendar;
+  DateTime startDateTime;
+  Duration duration;
```

Comment:
Why are we using Duration instead of Period here? Let's pick one and stick with it, it seems like we're doing a lot of unnecessary back and forth conversions between the two nearly equivalent types.

Reply:
I suggest actually using Duration everywhere. A package I just started using is using Duration too.